app/controllers/api/proxy.php

```diff
@@ -55,14 +55,19 @@
     ->inject('dbForPlatform')
     ->inject('dbForProject')
     ->action(function (string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject) {
+
         $mainDomain = System::getEnv('_APP_DOMAIN', '');
         if ($domain === $mainDomain) {
             throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your main domain to specific resource. Please use subdomain or a different domain.');
         }
 
-        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
-        if ($functionsDomain != '' && str_ends_with($domain, $functionsDomain)) {
-            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your functions domain or it\'s subdomain to specific resource. Please use different domain.');
+        $denyListString = System::getEnv('_APP_CUSTOM_DOMAIN_DENY_LIST', '');
+        $deniedDomains = array_map('trim', explode(',', $denyListString));
+
+        foreach ($deniedDomains as $deniedDomain) {
+            if (str_ends_with($domain, $deniedDomain)) {
+                throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your functions domain or its subdomain to a specific resource. Please use a different domain.');
+            }
```

Comment:
for backwards compatibility lets merge the two variables `_APP_DOMAIN_FUNCTIONS` and `_APP_CUSTOM_DOMAIN_DENY_LIST` so that we dont accidentally forget it

Reply:
Add _APP_CUSTOM_DOMAIN_DENY_LIST vars to variables.php and  inject it to the appwrite container